new error

sAyflort opened the pull request.

sAyflort committed:
new error

sAyflort committed:
new error

sAyflort committed:
new error

sAyflort committed:
new error 3

sAyflort committed:
Разработка передачи файлов по сети

sAyflort committed:
Добавление тестовых хендлеров

sAyflort committed:
- Полностью удалены предыдущие хендлеры.
- Вся кодировка работает на ObjectDecoder и ObjectEncoder
- Добавлена база данных с аутенфикацией
- Проект разделен на три модуля - клиент, сервер, классы-помощники

sAyflort committed:
- Запросы разбиты на несколько классов, имплементируемых от Request
- Добавлено обновление правой таблицы с серверными файлами
- Добавлена возможность копирования файлов с сервера на клиент

sAyflort committed:
- Добавлена логика удаления файла

BlackSan11 commented on filemanager-server/src/main/java/ru/geekbrains/cloud/ServerHandler.java:
```diff
@@ -0,0 +1,151 @@
+                    return;
+                }
+                if(sendFileRequest.getType() == DELETE_FILE) {
+                    String[] delPath = sendFileRequest.getDirectPath().split("\\\\");
```
здесь не проверяется атентифицирован ли пользователь, получается что любой может послать запрос на сервер и удалить файл

sAyflort committed:
- Добавлены наброски логгирования (будут доробатываться)
- Добавлена регистрация
- Добавлены информационные поля в окнах регистрации и аутенфикации
- Добавлены функции удаления файлов/пустых папок, создание папок
- Добавлен поиск файла по текущей и вложенных директориях

sAyflort committed:
класс, забытый в предыдущем коммите

sAyflort committed:
и еще один забытый файл

sAyflort committed:
commit

BlackSan11 commented on .idea/compiler.xml:
```diff
@@ -6,9 +6,12 @@
         <outputRelativeToContentRoot value="true" />
-        <module name="serverFileManager" />
         <module name="filemanager-client" />
+        <module name="filemanager-server" />
```
Общие и минимальные требования к оформлению pull request.

Стилистические требования:
1. Файлы .idea/ и .class не коммитятся, удалить из гитового индекса можно коммандной строкой(git rm --cached <PATH_TO_FILE>);

2. Необходимо выполнить код реформат(Ctrl + Alt + L), это позволит автоматически отформатировать код и привести его в стандартизированный вид.

3. Вложенность дальше 2-3 плохой стиль, очень ухудшает удобочитаемость кода:
Пример плохого кода:
```
if(...) {
	if(...) {
		if(...) {
			.....
		}
	}
}
```
Так лучше:
```
if(...) {
	otherActions();
	...
}

```
Семантические требования:

1. На серверной части необходимо выполнять проверку аутентификации пользователя перед выполнением каждого действия(запрос файла, запрос директорий на сервере, загрузка файла и т.д.), кроме действия на саму аутентификацию и регистрацию.

2. Большие файлы необходимо передавать частями, т.к. при не соблюдении этого требования будет черезмерно расходоваться оперативная память.

BlackSan11 commented on filemanager-server/src/main/java/ru/geekbrains/cloud/ServerHandler.java:
```diff
@@ -0,0 +1,163 @@
+            }
+            if(baseAuthService.isAuth(sendFileRequest.getLogin(), sendFileRequest.getPass())) {
+                LOGGER.info("Получен запрос: "+sendFileRequest.getType());
+                switch (sendFileRequest.getType()) {
```
Круто было бы здесь заюзать паттерн registry, сложить в хэшмеп все реквесты как ключ и лямбды как значения, в лямбде будет код обработки конкретного реквеста

BlackSan11 commented on filemanager-server/src/main/java/ru/geekbrains/cloud/dataBaseService/BaseAuthService.java:
```diff
@@ -0,0 +1,87 @@
+        try {
+            prSelectAuth.setString(1, login);
+            prSelectAuth.setString(2, pass);
+            ResultSet rs = prSelectAuth.executeQuery();
```
ResultSet это ресурс, его нужно обернуть в трай с ресурсами
